sdk/common/src/test/java/io/opentelemetry/sdk/resources/ResourceTest.java

```diff
@@ -330,4 +331,45 @@ public void toBuilder() {
     assertThat(newResource.getAttribute(stringKey("foo"))).isEqualTo("val");
     assertThat(newResource.getSchemaUrl()).isEqualTo("http://example.com");
   }
+
+  @Test
+  public void removeIf() {
+    ResourceBuilder builder = Resource.getDefault().toBuilder();
+    assertThat(builder.removeIf(unused -> true)).isEqualTo(builder);
+
+    assertThat(builder.removeIf(key -> key.getKey().equals("key1"))).isEqualTo(builder);
```

Comment:
These tests need to call `build()` on `ResourceBuilder` for a proper comparison. I.e.:
```
    assertThat(builder.removeIf(key -> key.getKey().equals("key1")).build()).isEqualTo(builder.build());
```